### DISCUSSION TITLE
Observables

### DISCUSSION THREAD

**Discussion by dennemark** [brianzinn/react-babylonjs#255] [Q&A]
Is my assumption right, that observables are updated properly?

https://github.com/brianzinn/react-babylonjs/blob/daa6605d1702f282365d797a7b1ab3bc0cb89db8/packages/react-babylonjs/src/UpdateInstance.ts#L54

When writing it as property, it is missing proper typing though. Would this be easy to solve?
Image shows typing when getting pointerDragBehaviour via create and the missing types when it is set as prop.

![image](https://user-images.githubusercontent.com/29654902/201050902-6bb1013e-7490-4ceb-9657-3ee20881a195.png)

**Comment by brianzinn** [comment]
Definitely possible and not overly difficult actually.  I just put `any` as I was adding it hastily and then I never went back and revisited the type safety of those.  Some of the code is here:
https://github.com/brianzinn/react-babylonjs/blob/master/packages/react-babylonjs/tools/generate-code.ts#L982

I noticed the type in Babylon.js they didn't extract and export a type, so it's really just a one-off:
https://github.com/BabylonJS/Babylon.js/blob/d25bc29091d47f51bd2f0f98fb0f16d25517675f/packages/dev/core/src/Behaviors/Meshes/pointerDragBehavior.ts#L86

I put in a breakpoint on the codegen and the type is fully available - I just never bothered to write it out to the JSX IntrinsicElement typings:
![image](https://user-images.githubusercontent.com/542756/201158016-8f28afcc-fec9-4baf-9bbb-96a17613c925.png)

Feel free to make an issue and that's not a bad idea, but it will take me a few weeks to get to it.  I'm really swamped with work lately unfortunately.  I definitely would like the typings to be stronger to help out and improve DX.

Thanks for the discussion!  Maybe it can be promoted somehow to be an issue.
